beacon-chain/blockchain/execution_engine.go

```diff
@@ -251,7 +251,7 @@ func (s *Service) optimisticCandidateBlock(ctx context.Context, blk interfaces.B
 		return nil
 	}
 
-	return errNotOptimisticCandidate
+	return invalidBlock{errNotOptimisticCandidate}
```

Comment:
If the block fails optimistic candidate check, the block should also be invalid

Reply:
I think this is not correct: A block may fail optimistic candidate because it is shallow across the fork, we can import it later when the chain advances further.